docs/src/modules/components/Link.tsx

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable jsx-a11y/anchor-has-content */
+import * as React from 'react';
+import clsx from 'clsx';
+import { useRouter } from 'next/router';
+import NextLink, { LinkProps as NextLinkProps } from 'next/link';
+import MuiLink, { LinkProps as MuiLinkProps } from '@material-ui/core/Link';
+import { useUserLanguage } from 'docs/src/modules/utils/i18n';
+
+type NextLinkComposedProps = Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, 'href'> &
+  Omit<NextLinkProps, 'href' | 'as'> & {
+    to: NextLinkProps['href'];
+    linkAs?: NextLinkProps['as'];
+    href?: any;
+  };
+
+export const NextLinkComposed = React.forwardRef<HTMLAnchorElement, NextLinkComposedProps>(
+  (props, ref) => {
+    const { to, linkAs, href, replace, scroll, passHref, shallow, prefetch, ...other } = props;
+
+    return (
+      <NextLink
+        href={to}
+        prefetch={prefetch}
+        as={linkAs}
+        replace={replace}
+        scroll={scroll}
+        shallow={shallow}
+        passHref={passHref}
+      >
+        <a ref={ref} {...other} />
+      </NextLink>
+    );
+  },
+);
+
+export type LinkProps = {
+  activeClassName?: string;
+  as?: NextLinkProps['as'];
+  noLinkStyle?: boolean;
+} & Omit<NextLinkComposedProps, 'to' | 'linkAs'> &
+  Omit<MuiLinkProps, 'href'>;
```

Comment:
If this can be solved with an `interface` please prefer an `interface` (https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections).

Reply:
I could migrate the first one